### ISSUE TITLE
(common) field in selectFieldset, which is enum, but has different options based on selection, how to 

### ISSUE THREAD

**Issue by sdetweil** [jsonform/jsonform#386]
I am trying to make a form over a list of different weather info providers. 

one common field NAME is a selection list (enum) 
BUT the  enum list is different between providers..

I know I can set the 'value' of a (string/num..) field in the form section,  but how would I define/use a enum there
here is the prototype
the selectFieldset works as expected

the field is the one I want to enhance, change to enum with no values in schema (and that is what I get in form, shows dropdown arrow, with no choices)
then add values in form 
```json
            "weatherEndpoint" :{
              "type": "string",
              "default":"/forecast"
            },
```
change to 
```json
            "weatherEndpoint" :{
              "type": "string",
             "enum":[]
            },
```
and then provide the enum values in the selectFieldset list of the form. 

I have tried enum, value, items, also tried all fields in enum (combo of all from all providers) 
and then a titlemap in form with "" as the value for not used values, BUT this makes an empty list, but the empties are selectable

```json
{
  "schema": {
    "weather": {
      "type": "object",
      "title": "properties for weather",
      "properties": {
        "module": {
          "type": "string",
          "title": "module",
          "default": "weather",
          "readonly": true
        },
        "disabled": {
          "type": "boolean",
          "title": "disabled",
          "default": false
        },
        "position": {
          "type": "string",
          "title": "module position",
          "readonly": "true"
        },
        "classes": {
          "type": "string",
          "title": "classes",
          "default": ""
        },
        "order": {
          "type": "string",
          "title": "order",
          "default": "*"
        },
        "inconfig": {
          "type": "string",
          "title": "inconfig",
          "default": "0"
        },
        "index": {
          "type": "integer"
        },
        "config": {
          "type": "object",
          "title": "config",
          "properties": {
            "weatherProvider": {
              "type": "string",
              "enum":[
                 "Dark Sky",
                 "Environment Canada",
                 "OpenWeatherMap",
                 "SMHI",
                 "UK Met Office (DataHub)",
                 "UK Met Office",
                 "Weatherbit",
                 "Weather.gov"
              ]
            },
            "apiBase": {
              "type":"string"              
            },
            "apiVersion": {
              "type":"string",
              "default":"2.5"
            },
            "weatherEndpoint" :{
              "type": "string",
              "default":"/forecast"
            },
            "apiKey": {
              "type":"string",
              "default":""
            },
            "apiSecret": {
              "type":"string",
              "default":""
            },
            "lat": {
              "type": "number",
              "default":0
            },
            "lon": {
              "type": "number",
              "default":0
            },
            "siteCode": {
              "type":"string",
              "default":"s1234567"
            },
            "provCode": {
              "type":"string",
              "default":"ON"
            },
            "locationID": {
              "type":"string",
              "default":""
            },
            "location": {
              "type":"string",
              "default":""
            },
            "precipitationValue": {
              "type":"string",
              "default":"pmedian"
            },
            "windUnits": {
              "type":"string",
              "enum":[
                "mph",
                "kph"
              ]
            },
            "roundTemp": {
              "type": "boolean"
            },
            "type": {
              "type": "string"
            },
            "units": {
              "type": "string"
            },
            "useKmh": {
              "type": "boolean"
            },
            "tempUnits": {
              "type": "string"
            },
            "updateInterval": {
              "type": "integer"
            },
            "animationSpeed": {
              "type": "integer"
            },
            "timeFormat": {
              "type": "integer"
            },
            "showPeriod": {
              "type": "boolean"
            },
            "showPeriodUpper": {
              "type": "boolean"
            },
            "showWindDirection": {
              "type": "boolean"
            },
            "showWindDirectionAsArrow": {
              "type": "boolean"
            },
            "useBeaufort": {
              "type": "boolean"
            },
            "lang": {
              "type": "string"
            },
            "showHumidity": {
              "type": "boolean"
            },
            "showSun": {
              "type": "boolean"
            },
            "degreeLabel": {
              "type": "string"
            },
            "decimalSymbol": {
              "type": "string"
            },
            "showIndoorTemperature": {
              "type": "boolean"
            },
            "showIndoorHumidity": {
              "type": "boolean"
            },
            "maxNumberOfDays": {
              "type": "integer"
            },
            "maxEntries": {
              "type": "integer"
            },
            "fade": {
              "type": "boolean"
            },
            "fadePoint": {
              "type": "integer"
            },
            "initialLoadDelay": {
              "type": "integer"
            },
            "appendLocationNameToHeader": {
              "type": "boolean"
            },
            "calendarClass": {
              "type": "string"
            },
            "tableClass": {
              "type": "string"
            },
            "onlyTemp": {
              "type": "boolean"
            },
            "showPrecipitationAmount": {
              "type": "boolean"
            },
            "colored": {
              "type": "boolean"
            },
            "showFeelsLike": {
              "type": "boolean"
            }
          }
        }
      }
    }
  },
  "form": [
    {
      "key": "weather.disabled",
      "htmlClass": "disabled_checkbox",
      "description": "when checked the module will not be used by MagicMirror<br> but will remain in config.js if already present"
    },
    {
      "key": "weather.position",
      "description": "use Module Positions section below to set or change"
    },
    {
      "key": "weather.classes",
      "description": "css classes to use for this module, beyond what MM provides"
    },
    {
      "key": "weather.order",
      "type": "hidden"
    },
    {
      "key": "weather.inconfig",
      "type": "hidden"
    },
    {
      "key": "weather.index",
      "type": "hidden"
    },
    {
      "type": "section",
      "title": "config",
      "items": [
        {
          "type": "selectfieldset",
          "title": "weatherProvider",
          "items": [
            {
              "type":"section",
              "title":"Dark Sky",
              "items":[
                {
                  "title":"apiBase",
                  "key":"weather.config.apiBase",
                  "value": "https://cors-anywhere.herokuapp.com/https://api.darksky.net"                  
                },
                {
                  "title":"weatherEndpoint",                
                  "key":"weather.config.weatherEndpoint"
                },
                {
                  "title":"apiKey",                    
                  "key":"weather.config.apiKey"
                },
                {
                  "title":"lat",                  
                  "key":"weather.config.lat"
                },
                {
                  "title":"lon",                  
                  "key":"weather.config.lon"
                }
              ]
            },
            {
              "type":"section",
              "title": "Environment Canada",
              "items":[
                {
                  "title":"siteCode",    
                  "key":"weather.config.siteCode"
                },
                {
                  "title":"provCode",                 
                  "key":"weather.config.provCode"
                }
              ]
            },          
            {
              "type":"section",
              "title": "OpenWeatherMap",
              "items":[
                {
                  "title":"apiVersion",
                  "key":"weather.config.apiVersion"
                },                
                {
                  "title":"apiBase",
                  "key":"weather.config.apiBase",
                  "value": "https://api.openweathermap.org/data/"
                },
                {
                  "title":"weatherEndpoint",                
                  "key":"weather.config.weatherEndpoint"
                },
                {
                  "title":"apiKey",                    
                  "key":"weather.config.apiKey"
                },
                {
                  "title":"lat",                  
                  "key":"weather.config.lat"
                },
                {
                  "title":"lon",                  
                  "key":"weather.config.lon"
                },
                {
                  "title":"locationID",                  
                  "key":"weather.config.locationID"
                },  
                {                  
                  "title":"location",                  
                  "key":"weather.config.location"
                }
              ]
            },
            {
              "type":"section",
              "title": "smhi",
              "items":[
                {
                  "title":"lat",                  
                  "key":"weather.config.lat"
                },
                {
                  "title":"lon",                  
                  "key":"weather.config.lon"
                },
                {
                  "title":"precipitationValue",                  
                  "key":"weather.config.precipitationValue"
                }                 
              ]
            },
            {
              "type":"section",
              "title": "UK Met Office (DataHub)",
              "items":[
                {
                  "title":"apiBase",
                  "key":"weather.config.apiBase",
                  "value": "https://api-metoffice.apiconnect.ibmcloud.com/metoffice/production/v0/forecasts/point/"
                },
                {
                  "title":"apiKey",                    
                  "key":"weather.config.apiKey"
                }, 
                {
                  "title":"apiSecret",                    
                  "key":"weather.config.apiSecret"
                },                               
                {
                  "title":"lat",                  
                  "key":"weather.config.lat"
                },
                {
                  "title":"lon",                  
                  "key":"weather.config.lon"
                },
                {
                  "title":"windUnits",                  
                  "key":"weather.config.windUnits"
                }                 
              ]
            },         
            {
              "type":"section",
              "title": "UK Met Office",
              "items":[
                {
                  "title":"apiBase",
                  "key":"weather.config.apiBase",
                  "value": "http://datapoint.metoffice.gov.uk/public/data/val/wxfcs/all/json/"
                },                
                {
                  "title":"apiKey",                    
                  "key":"weather.config.apiKey"
                },
                {
                  "title":"locationID",                  
                  "key":"weather.config.locationID"
                }
              ]
            },
            {
              "type":"section",
              "title": "Weatherbit",
              "items":[               
                {
                  "title":"apiBase",
                  "key":"weather.config.apiBase",
                  "value": "https://api.weatherbit.io/v2.0"                   
                },
                {
                  "title":"weatherEndpoint",                
                  "key":"weather.config.weatherEndpoint"                 
                },
                {
                  "title":"apiKey",                    
                  "key":"weather.config.apiKey"
                },
                {
                  "title":"lat",                  
                  "key":"weather.config.lat"
                },
                {
                  "title":"lon",                  
                  "key":"weather.config.lon"
                }
              ]
            },    
            {
              "type":"section",
              "title": "Weather.gov",
              "items":[               
                {
                  "title":"apiBase",
                  "key":"weather.config.apiBase",
                  "value": "https://api.weather.gov/points/",
                },
                {
                  "title":"weatherEndpoint",                
                  "key":"weather.config.weatherEndpoint"
                },
                {
                  "title":"lat",                  
                  "key":"weather.config.lat"
                },
                {
                  "title":"lon",                  
                  "key":"weather.config.lon"
                }
              ]
            },            
          ]
        },
        {
          "title": "roundTemp",
          "key": "weather.config.roundTemp"
        },
        {
          "title": "type",
          "key": "weather.config.type"
        },
        {
          "title": "units",
          "key": "weather.config.units"
        },
        {
          "title": "useKmh",
          "key": "weather.config.useKmh"
        },
        {
          "title": "tempUnits",
          "key": "weather.config.tempUnits"
        },
        {
          "title": "windUnits",
          "key": "weather.config.windUnits"
        },
        {
          "title": "updateInterval",
          "key": "weather.config.updateInterval"
        },
        {
          "title": "animationSpeed",
          "key": "weather.config.animationSpeed"
        },
        {
          "title": "timeFormat",
          "key": "weather.config.timeFormat"
        },
        {
          "title": "showPeriod",
          "key": "weather.config.showPeriod"
        },
        {
          "title": "showPeriodUpper",
          "key": "weather.config.showPeriodUpper"
        },
        {
          "title": "showWindDirection",
          "key": "weather.config.showWindDirection"
        },
        {
          "title": "showWindDirectionAsArrow",
          "key": "weather.config.showWindDirectionAsArrow"
        },
        {
          "title": "useBeaufort",
          "key": "weather.config.useBeaufort"
        },
        {
          "title": "lang",
          "key": "weather.config.lang"
        },
        {
          "title": "showHumidity",
          "key": "weather.config.showHumidity"
        },
        {
          "title": "showSun",
          "key": "weather.config.showSun"
        },
        {
          "title": "degreeLabel",
          "key": "weather.config.degreeLabel"
        },
        {
          "title": "decimalSymbol",
          "key": "weather.config.decimalSymbol"
        },
        {
          "title": "showIndoorTemperature",
          "key": "weather.config.showIndoorTemperature"
        },
        {
          "title": "showIndoorHumidity",
          "key": "weather.config.showIndoorHumidity"
        },
        {
          "title": "maxNumberOfDays",
          "key": "weather.config.maxNumberOfDays"
        },
        {
          "title": "maxEntries",
          "key": "weather.config.maxEntries"
        },
        {
          "title": "fade",
          "key": "weather.config.fade"
        },
        {
          "title": "fadePoint",
          "key": "weather.config.fadePoint"
        },
        {
          "title": "initialLoadDelay",
          "key": "weather.config.initialLoadDelay"
        },
        {
          "title": "appendLocationNameToHeader",
          "key": "weather.config.appendLocationNameToHeader"
        },
        {
          "title": "calendarClass",
          "key": "weather.config.calendarClass"
        },
        {
          "title": "tableClass",
          "key": "weather.config.tableClass"
        },
        {
          "title": "onlyTemp",
          "key": "weather.config.onlyTemp"
        },
        {
          "title": "showPrecipitationAmount",
          "key": "weather.config.showPrecipitationAmount"
        },
        {
          "title": "colored",
          "key": "weather.config.colored"
        },
        {
          "title": "showFeelsLike",
          "key": "weather.config.showFeelsLike"
        }
      ]
    }
  ]
}
```

**Comment by sdetweil** [comment]
of the 8 providers

2 use just /forecast
1 uses just /current
and 1 uses
/forecast
/forcast/daily
/onecall
/weather

the other 4 don't use this field (and the selectFieldset works there

**Comment by sdetweil** [comment]
so, after fiddling, this 'works' from the presentation approach , I don't know if it create the right data output
the field is a string 
```
            "weatherEndpoint" :{
              "type": "string"
            },
```
each selectFieldset entry provides a different 'value'
and for those that need a selection list
use  items, with title and value...

```json
{
  "schema": {
    "weather": {
      "type": "object",
      "title": "properties for weather",
      "properties": {
        "module": {
          "type": "string",
          "title": "module",
          "default": "weather",
          "readonly": true
        },
        "disabled": {
          "type": "boolean",
          "title": "disabled",
          "default": false
        },
        "position": {
          "type": "string",
          "title": "module position",
          "readonly": "true"
        },
        "classes": {
          "type": "string",
          "title": "classes",
          "default": ""
        },
        "order": {
          "type": "string",
          "title": "order",
          "default": "*"
        },
        "inconfig": {
          "type": "string",
          "title": "inconfig",
          "default": "0"
        },
        "index": {
          "type": "integer"
        },
        "config": {
          "type": "object",
          "title": "config",
          "properties": {
            "weatherProvider": {
              "type": "string",
              "enum":[
                 "Dark Sky",
                 "Environment Canada",
                 "OpenWeatherMap",
                 "SMHI",
                 "UK Met Office (DataHub)",
                 "UK Met Office",
                 "Weatherbit",
                 "Weather.gov"
              ]
            },
            "apiBase": {
              "type":"string"              
            },
            "apiVersion": {
              "type":"string",
              "default":"2.5"
            },
            "weatherEndpoint" :{
              "type": "string"
            },
            "apiKey": {
              "type":"string",
              "default":""
            },
            "apiSecret": {
              "type":"string",
              "default":""
            },
            "lat": {
              "type": "number",
              "default":0
            },
            "lon": {
              "type": "number",
              "default":0
            },
            "siteCode": {
              "type":"string",
              "default":"s1234567"
            },
            "provCode": {
              "type":"string",
              "default":"ON"
            },
            "locationID": {
              "type":"string",
              "default":""
            },
            "location": {
              "type":"string",
              "default":""
            },
            "precipitationValue": {
              "type":"string",
              "default":"pmedian"
            },
            "windUnits": {
              "type":"string",
              "enum":[
                "mph",
                "kph"
              ]
            },
            "roundTemp": {
              "type": "boolean"
            },
            "type": {
              "type": "string"
            },
            "units": {
              "type": "string"
            },
            "useKmh": {
              "type": "boolean"
            },
            "tempUnits": {
              "type": "string"
            },
            "updateInterval": {
              "type": "integer"
            },
            "animationSpeed": {
              "type": "integer"
            },
            "timeFormat": {
              "type": "integer"
            },
            "showPeriod": {
              "type": "boolean"
            },
            "showPeriodUpper": {
              "type": "boolean"
            },
            "showWindDirection": {
              "type": "boolean"
            },
            "showWindDirectionAsArrow": {
              "type": "boolean"
            },
            "useBeaufort": {
              "type": "boolean"
            },
            "lang": {
              "type": "string"
            },
            "showHumidity": {
              "type": "boolean"
            },
            "showSun": {
              "type": "boolean"
            },
            "degreeLabel": {
              "type": "string"
            },
            "decimalSymbol": {
              "type": "string"
            },
            "showIndoorTemperature": {
              "type": "boolean"
            },
            "showIndoorHumidity": {
              "type": "boolean"
            },
            "maxNumberOfDays": {
              "type": "integer"
            },
            "maxEntries": {
              "type": "integer"
            },
            "fade": {
              "type": "boolean"
            },
            "fadePoint": {
              "type": "integer"
            },
            "initialLoadDelay": {
              "type": "integer"
            },
            "appendLocationNameToHeader": {
              "type": "boolean"
            },
            "calendarClass": {
              "type": "string"
            },
            "tableClass": {
              "type": "string"
            },
            "onlyTemp": {
              "type": "boolean"
            },
            "showPrecipitationAmount": {
              "type": "boolean"
            },
            "colored": {
              "type": "boolean"
            },
            "showFeelsLike": {
              "type": "boolean"
            }
          }
        }
      }
    }
  },
  "form": [
    {
      "key": "weather.disabled",
      "htmlClass": "disabled_checkbox",
      "description": "when checked the module will not be used by MagicMirror<br> but will remain in config.js if already present"
    },
    {
      "key": "weather.position",
      "description": "use Module Positions section below to set or change"
    },
    {
      "key": "weather.classes",
      "description": "css classes to use for this module, beyond what MM provides"
    },
    {
      "key": "weather.order",
      "type": "hidden"
    },
    {
      "key": "weather.inconfig",
      "type": "hidden"
    },
    {
      "key": "weather.index",
      "type": "hidden"
    },
    {
      "type": "section",
      "title": "config",
      "items": [
        {
          "type": "selectfieldset",
          "title": "weatherProvider",
          "items": [
            {
              "type":"section",
              "title":"Dark Sky",
              "items":[
                {
                  "title":"apiBase",
                  "key":"weather.config.apiBase",
                  "value": "https://cors-anywhere.herokuapp.com/https://api.darksky.net"                  
                },
                {
                  "title":"weatherEndpoint",
                  "value":"/forecast",
                  "key":"weather.config.weatherEndpoint",
                  "readonly":"true"
                },
                {
                  "title":"apiKey",                    
                  "key":"weather.config.apiKey"
                },
                {
                  "title":"lat",                  
                  "key":"weather.config.lat"
                },
                {
                  "title":"lon",                  
                  "key":"weather.config.lon"
                }
              ]
            },
            {
              "type":"section",
              "title": "Environment Canada",
              "items":[
                {
                  "title":"siteCode",    
                  "key":"weather.config.siteCode"
                },
                {
                  "title":"provCode",                 
                  "key":"weather.config.provCode"
                }
              ]
            },          
            {
              "type":"section",
              "title": "OpenWeatherMap",
              "items":[
                {
                  "title":"apiVersion",
                  "key":"weather.config.apiVersion"
                },                
                {
                  "title":"apiBase",
                  "key":"weather.config.apiBase",
                  "value": "https://api.openweathermap.org/data/"
                },
                {
                  "title":"weatherEndpoint",                
                  "key":"weather.config.weatherEndpoint",
                  "type":"selectfieldset",
                  "items":[
                    {
                      "title":"/forecast",
                      "value":"/forecast"
                    },
                    {
                      "title":"/forecast/daily",
                      "value":"/forecast/daily"
                    },
                    {
                      "title":"/weather",
                      "value":"/weather"
                    },
                    {
                      "title":"/onecall",
                      "value":"/onecall"
                    }
                  ]
                },
                {
                  "title":"apiKey",                    
                  "key":"weather.config.apiKey"
                },
                {
                  "title":"lat",                  
                  "key":"weather.config.lat"
                },
                {
                  "title":"lon",                  
                  "key":"weather.config.lon"
                },
                {
                  "title":"locationID",                  
                  "key":"weather.config.locationID"
                },  
                {                  
                  "title":"location",                  
                  "key":"weather.config.location"
                }
              ]
            },
            {
              "type":"section",
              "title": "smhi",
              "items":[
                {
                  "title":"lat",                  
                  "key":"weather.config.lat"
                },
                {
                  "title":"lon",                  
                  "key":"weather.config.lon"
                },
                {
                  "title":"precipitationValue",                  
                  "key":"weather.config.precipitationValue"
                }                 
              ]
            },
            {
              "type":"section",
              "title": "UK Met Office (DataHub)",
              "items":[
                {
                  "title":"apiBase",
                  "key":"weather.config.apiBase",
                  "value": "https://api-metoffice.apiconnect.ibmcloud.com/metoffice/production/v0/forecasts/point/"
                },
                {
                  "title":"apiKey",                    
                  "key":"weather.config.apiKey"
                }, 
                {
                  "title":"apiSecret",                    
                  "key":"weather.config.apiSecret"
                },                               
                {
                  "title":"lat",                  
                  "key":"weather.config.lat"
                },
                {
                  "title":"lon",                  
                  "key":"weather.config.lon"
                },
                {
                  "title":"windUnits",                  
                  "key":"weather.config.windUnits"
                }                 
              ]
            },         
            {
              "type":"section",
              "title": "UK Met Office",
              "items":[
                {
                  "title":"apiBase",
                  "key":"weather.config.apiBase",
                  "value": "http://datapoint.metoffice.gov.uk/public/data/val/wxfcs/all/json/"
                },                
                {
                  "title":"apiKey",                    
                  "key":"weather.config.apiKey"
                },
                {
                  "title":"locationID",                  
                  "key":"weather.config.locationID"
                }
              ]
            },
            {
              "type":"section",
              "title": "Weatherbit",
              "items":[               
                {
                  "title":"apiBase",
                  "key":"weather.config.apiBase",
                  "value": "https://api.weatherbit.io/v2.0"                   
                },
                {
                  "title":"weatherEndpoint",                
                  "value":"/current",
                  "key":"weather.config.weatherEndpoint",
                  "readonly":"true"
                },
                {
                  "title":"apiKey",                    
                  "key":"weather.config.apiKey"
                },
                {
                  "title":"lat",                  
                  "key":"weather.config.lat"
                },
                {
                  "title":"lon",                  
                  "key":"weather.config.lon"
                }
              ]
            },    
            {
              "type":"section",
              "title": "Weather.gov",
              "items":[               
                {
                  "title":"apiBase",
                  "key":"weather.config.apiBase",
                  "value": "https://api.weather.gov/points/",
                },
                {
                  "title":"weatherEndpoint",                
                  "value":"/forecast",
                  "key":"weather.config.weatherEndpoint",
                  "readonly":"true"
                },
                {
                  "title":"lat",                  
                  "key":"weather.config.lat"
                },
                {
                  "title":"lon",                  
                  "key":"weather.config.lon"
                }
              ]
            },            
          ]
        },
        {
          "title": "roundTemp",
          "key": "weather.config.roundTemp"
        },
        {
          "title": "type",
          "key": "weather.config.type"
        },
        {
          "title": "units",
          "key": "weather.config.units"
        },
        {
          "title": "useKmh",
          "key": "weather.config.useKmh"
        },
        {
          "title": "tempUnits",
          "key": "weather.config.tempUnits"
        },
        {
          "title": "windUnits",
          "key": "weather.config.windUnits"
        },
        {
          "title": "updateInterval",
          "key": "weather.config.updateInterval"
        },
        {
          "title": "animationSpeed",
          "key": "weather.config.animationSpeed"
        },
        {
          "title": "timeFormat",
          "key": "weather.config.timeFormat"
        },
        {
          "title": "showPeriod",
          "key": "weather.config.showPeriod"
        },
        {
          "title": "showPeriodUpper",
          "key": "weather.config.showPeriodUpper"
        },
        {
          "title": "showWindDirection",
          "key": "weather.config.showWindDirection"
        },
        {
          "title": "showWindDirectionAsArrow",
          "key": "weather.config.showWindDirectionAsArrow"
        },
        {
          "title": "useBeaufort",
          "key": "weather.config.useBeaufort"
        },
        {
          "title": "lang",
          "key": "weather.config.lang"
        },
        {
          "title": "showHumidity",
          "key": "weather.config.showHumidity"
        },
        {
          "title": "showSun",
          "key": "weather.config.showSun"
        },
        {
          "title": "degreeLabel",
          "key": "weather.config.degreeLabel"
        },
        {
          "title": "decimalSymbol",
          "key": "weather.config.decimalSymbol"
        },
        {
          "title": "showIndoorTemperature",
          "key": "weather.config.showIndoorTemperature"
        },
        {
          "title": "showIndoorHumidity",
          "key": "weather.config.showIndoorHumidity"
        },
        {
          "title": "maxNumberOfDays",
          "key": "weather.config.maxNumberOfDays"
        },
        {
          "title": "maxEntries",
          "key": "weather.config.maxEntries"
        },
        {
          "title": "fade",
          "key": "weather.config.fade"
        },
        {
          "title": "fadePoint",
          "key": "weather.config.fadePoint"
        },
        {
          "title": "initialLoadDelay",
          "key": "weather.config.initialLoadDelay"
        },
        {
          "title": "appendLocationNameToHeader",
          "key": "weather.config.appendLocationNameToHeader"
        },
        {
          "title": "calendarClass",
          "key": "weather.config.calendarClass"
        },
        {
          "title": "tableClass",
          "key": "weather.config.tableClass"
        },
        {
          "title": "onlyTemp",
          "key": "weather.config.onlyTemp"
        },
        {
          "title": "showPrecipitationAmount",
          "key": "weather.config.showPrecipitationAmount"
        },
        {
          "title": "colored",
          "key": "weather.config.colored"
        },
        {
          "title": "showFeelsLike",
          "key": "weather.config.showFeelsLike"
        }
      ]
    }
  ]
}
```

**Comment by sdetweil** [comment]
so close...!!!
```
         {
                  "title":"weatherEndpoint",                
                  "key":"weather.config.weatherEndpoint",
                  "type":"selectfieldset",
                  "items":[
                    {
                      "title":"/forecast",
                      "value":"/forecast"
                    },
                    {
                      "title":"/forecast/daily",
                      "value":"/forecast/daily"
                    },
                    {
                      "title":"/weather",
                      "value":"/weather"
                    },
                    {
                      "title":"/onecall",
                      "value":"/onecall"
                    }
                  ]
                },
```
returns 0 from the form submission, regardless of which item is selected in the select list dropdown

looking at the html , the select list has the 1st item 'selected', while changing the list item, changed the 'active' class
![Screenshot at 2021-11-11 22-14-50](https://user-images.githubusercontent.com/1498142/141408505-fbc632c0-a8c9-402d-b952-fed0d661e0f1.png)

**Comment by sdetweil** [comment]
the on change handler for the selectfieldset , does this 
```
    'onInsert': function (evt, node) {
      $(node.el).find('select.nav').first().on('change', function (evt) {
        var $option = $(this).find('option:selected');
        $(node.el).find('input[type="hidden"]').first().val($option.attr('value'));
      });
```

find the option that is 'selected'
then use the value of that node to set the value of the hidden input field, which is the key field for this select 

but selected never changes..   but active does

if the code is changed like this 
```
    'onInsert': function (evt, node) {
      $(node.el).find('select.nav').first().on('change', function (evt) {
        var $option = $(this).find('option.active');
        $(node.el).find('input[type="hidden"]').first().val($option.attr('value'));
      });
```

the select onChange is not firing.. it is set up (code stops on the call to setup the change handler.. )
something is handling the active class movement..  is that not propgating the change event?

**Comment by sdetweil** [comment]
ok, the select triggering problem is that the formelement/el is not pointing to the right node on the nested selectfieldset. 
on the 1st outer selectlist at the time to call is made to setup the onchange handler 
![node_el_1st](https://user-images.githubusercontent.com/1498142/141533815-6e49414d-998f-4c8d-9c34-efd325bbe800.png)

note here el  is pointing at fieldset.tab-container

![after_find_1st](https://user-images.githubusercontent.com/1498142/141533813-1f97a62b-146c-4151-a3e5-d241c2d980c1.png)

after fine of select.nav
there are two selects found, the code then pics the 1st. 

now the seconds selectfieldset 
![node_el_2nd](https://user-images.githubusercontent.com/1498142/141533917-293c6911-529b-44ea-a14c-14a5827cfaed.png)

note here el  is pointing at div.controls

![after_find_2nd](https://user-images.githubusercontent.com/1498142/141533919-98fd30d3-dac6-4aad-9562-33a414bc0f15.png)

and after find of select.nav
no items found, so the add of the onchange handler fails


after moving the items in the list around, it works.. BUT, this calls out an implementation problem

there is an assumption that the section is ACTIVE at startup time, and the original design had this section as third in the list, meaning NOT active at startup., but then later is made active (due to the form data), but the on change handler is not reset when the new item comes active


basically  instead of onInsert it needs OnChange to look for any select.nav children and set their  on change handler 
and UNHOOK any onchange handler no longer in context.

**Comment by sdetweil** [comment]
added pr #391 to enhance selectFieldset onchange handler, and onInsert function
